library/general/noupdateserviceaccount/samples/noupdateserviceaccount/example_allowed.yaml

```diff
@@ -0,0 +1,65 @@
+kind: ServiceAccount
```

Comment:
I'm not really sure how these tests work. Do we have an emulated kubernetes API such that we can test mutations of existing resources, or is it a stateless test?

Reply:
Unfortunately the tests are essentially stateless (only CREATE is tested). For our local tests, we are looking at ways to provide a full K8s AdmissionReview object so that things like "only-on-update" can be appropriately tested.

Generally edge-based enforcement is discouraged if there is an alternative (only works at request time, does not work for audit), but because this is specifically testing changes to the service account it seems reasonable.

Reply:
Got it - what's the best thing to do here? Just comment out the "disallowed" example from `suite.yaml`?

For level-triggered audits like this, you _might_ be able to extract past revisions of Deployments, etc. They have to be stored _somewhere_ since there's a rollback option: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#checking-rollout-history-of-a-deployment

Reply:
Yeah, commenting out for now seems like the best course.

re: level-triggered audits... interesting. I wonder where this data lives? Currently we are scraping the objects as they exist on the cluster.

Reply:
Ack, done.